admin/services/pupil-access-arrangements.service.js

```diff
@@ -30,4 +30,28 @@ pupilAccessArrangementsService.getPupils = async (dfeNumber) => {
   return pupilIdentificationFlag.addIdentificationFlags(pupilsWithAccessArrangements)
 }
 
+/**
+ * Returns pupils edit form data
+ * @param {String} urlSlug
+ * @returns {Object}
+ */
+pupilAccessArrangementsService.getPupilEditFormData = async (urlSlug) => {
+  if (!urlSlug) {
+    throw new Error('Pupil url slug is not provided')
+  }
+  const pupilAccessArrangementsData = await pupilAccessArrangementsDataService.sqlFindAccessArrangementsByUrlSlug(urlSlug)
+  const formData = pupilAccessArrangementsData.reduce((acc, item) => {
+    acc.pupilUrlSlug = !acc.pupilUrlSlug ? item.urlSlug : acc.pupilUrlSlug
+    acc.foreName = !acc.foreName ? item.foreName : acc.foreName
+    acc.lastName = !acc.lastName ? item.lastName : acc.lastName
+    acc.inputAssistanceInformation = !acc.inputAssistanceInformation ? item.inputAssistanceInformation : acc.inputAssistanceInformation
+    acc.questionReaderOtherInformation = !acc.questionReaderOtherInformation ? item.questionReaderOtherInformation : acc.questionReaderOtherInformation
+    acc.questionReaderReason = !acc.questionReaderReason ? item.questionReaderReasonCode : acc.questionReaderReason
```

Comment:
Can these be something like `acc.pupilUrlSlug = acc.pupilUrlSlug || item.urlSlug` instead, if I got that right? i.e if `acc.pupilUrlSlug` is undefined, it will be `item.urlSlug`, and if that is undefined too, the final result is undefined (although I don’t think that will happen?) 🤔

Reply:
@danwilliams valid point the ternary is completely unnecessary.
It's updated now and yes it can't be go into undefined.